Mathlib/CategoryTheory/Filtered.lean

```diff
@@ -487,6 +488,146 @@ theorem tulip {j₁ j₂ j₃ k₁ k₂ l : C} (f₁ : j₁ ⟶ k₁) (f₂ : j
+  fun x => (bundledAbstractFilteredClosure f x.1).2 x.2
+
+end FilteredClosureSmall
+
+theorem small_fullSubcategory_filteredClosure :
+    Small.{max v w} (FullSubcategory (FilteredClosure f)) := by
+  refine' small_of_injective_of_exists (FilteredClosureSmall.abstractFilteredClosureRealization f)
+    FullSubcategory.ext _
+  rintro ⟨j, h⟩
+  induction h with
+  | base x => exact ⟨⟨0, ⟨x⟩⟩, rfl⟩
+  | max hj₁ hj₂ ih ih' =>
+    rcases ih with ⟨⟨n, x⟩, rfl⟩
+    rcases ih' with ⟨⟨m, y⟩, rfl⟩
+    refine' ⟨⟨(Max.max n m).succ, FilteredClosureSmall.InductiveStep.max _ _ x y⟩, rfl⟩
+    all_goals apply Nat.lt_succ_of_le
+    exacts [Nat.le_max_left _ _, Nat.le_max_right _ _]
+  | coeq hj₁ hj₂ g g' ih ih' =>
+    rcases ih with ⟨⟨n, x⟩, rfl⟩
+    rcases ih' with ⟨⟨m, y⟩, rfl⟩
+    refine' ⟨⟨(Max.max n m).succ, FilteredClosureSmall.InductiveStep.coeq _ _ x y g g'⟩, rfl⟩
+    all_goals apply Nat.lt_succ_of_le
+    exacts [Nat.le_max_left _ _, Nat.le_max_right _ _]
+
+instance : EssentiallySmall.{max v w} (FullSubcategory (FilteredClosure f)) :=
+  have : LocallySmall.{max v w} (FullSubcategory (FilteredClosure f)) := locallySmall_max.{w, v, u}
+  have := small_fullSubcategory_filteredClosure f
+  essentiallySmall_of_small_of_locallySmall _
+
+end FilteredClosure
+
+section
+
+variable {C}
+variable [IsFilteredOrEmpty C] {D : Type u₁} [Category.{v₁} D] (F : D ⥤ C)
+
+/-- Every functor from a small category to a filtered category factors fully faithfully through a
+    small filtered category. This is that category. -/
+def SmallFilteredIntermediate : Type (max u₁ v) :=
+  SmallModel.{max u₁ v} (FullSubcategory (FilteredClosure F.obj))
+
+noncomputable instance : SmallCategory (SmallFilteredIntermediate F) :=
+  show SmallCategory (SmallModel (FullSubcategory (FilteredClosure F.obj))) from inferInstance
+
+/-- The first part of a factoring of a functor from a small category to a filtered category through
+    a small filtered category. -/
+noncomputable def smallFilteredIntermediateFactoring : D ⥤ SmallFilteredIntermediate F :=
+  FullSubcategory.lift _ F FilteredClosure.base ⋙ (equivSmallModel _).functor
+
+/-- The second, fully faithful part of a factoring of a functor from a small category to a filtered
+    category through a small filtered category. -/
+noncomputable def smallFilteredIntermediateInclusion : SmallFilteredIntermediate F ⥤ C :=
+  (equivSmallModel _).inverse ⋙ fullSubcategoryInclusion _
+
+instance : Faithful (smallFilteredIntermediateInclusion F) :=
+  show Faithful ((equivSmallModel _).inverse ⋙ fullSubcategoryInclusion _) from Faithful.comp _ _
+
+noncomputable instance : Full (smallFilteredIntermediateInclusion F) :=
+  show Full ((equivSmallModel _).inverse ⋙ fullSubcategoryInclusion _) from Full.comp _ _
+
+/-- The factorization through a small filtered category is in fact a factorization, up to natural
+    isomorphism. -/
+noncomputable def smallFilteredIntermediateFactoringCompSmallFilteredIntermediateInclusion :
```

Comment:
If you rename the above two defs to `factoring` and `inclusion` and put them in the namespace `SmallFilteredIntermediate`, then this def can get a much shorter name.